src/Hooks/usePokemon.js

```diff
@@ -0,0 +1,38 @@
+import { fetchPokemon, fetchPokeTypes } from '../services/fetch-utils';
+import { useState, useEffect } from 'react';
+
+export default function usePokemon() {
+  const [pokemonList, setPokemonList] = useState([]);
+  const [type, setType] = useState('All');
+  const [typeList, setTypeList] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [inputValue, setInputValue] = useState('');
+  useEffect(() => {
+    const loadData = async () => {
+      try {
+        const data = await fetchPokemon(type);
+        setPokemonList(data);
+        setLoading(false);
+      } catch (e) {
+    // eslint-disable-next-line no-console
+        console.error(e);
+      }
+    };
+    loadData();
+  }, [type]);
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        const data = await fetchPokeTypes();
+        setTypeList(data);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.error(e);
```

Comment:
Would rather see some error state here that gets passed to a component and rendered for the user to see